From de17fdf789fec7f74987be894b9ed77f635ac038 Mon Sep 17 00:00:00 2001
From: Wincent Colaiuta <win@wincent.com>
Date: Wed, 28 May 2008 16:36:57 +0200
Subject: [PATCH 2/2] Support gem dependencies which need "ext" in the load path

Previously we only added the "lib" subdirectory to the load path when
setting up gem dependencies for frozen gems. Now we add the "ext"
subdirectory as well for those gems which have compiled C extensions
as well.

Signed-off-by: Wincent Colaiuta <win@wincent.com>
---
 railties/lib/rails/gem_dependency.rb |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/railties/lib/rails/gem_dependency.rb b/railties/lib/rails/gem_dependency.rb
index 2034841..703e282 100644
--- a/railties/lib/rails/gem_dependency.rb
+++ b/railties/lib/rails/gem_dependency.rb
@@ -32,6 +32,8 @@ module Rails
         gem *args
       else
         $LOAD_PATH << File.join(unpacked_paths.first, 'lib')
+        ext = File.join(unpacked_paths.first, 'ext')
+        $LOAD_PATH << ext if File.exist?(ext)
         @frozen = true
       end
       @load_paths_added = true
-- 
1.5.5.3.dirty

